### PR TITLE
stuff

### PR DESCRIPTION
**Pull request by tomatosource**
honestly this was nice.

the main/do pattern is def something i only do in side projects but yeah it makes refactoring stuff later way easier - never panic (except in that one spot).

few comments on some ideas to play with, be real nice to do the sig stuff, the channel setup to do that properly is interesting.

go@euc when.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -6,60 +6,126 @@ import (
...
+	if err := do(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func do() error {
 	ex, err := os.Executable()
 	if err != nil {
-		panic(err)
+		return errors.Wrap(err, "getting executable")
```
Comment (tomatosource) [inline main.go:33]: this is a convention from real world, but i use in side projects now because of how useful it is.

Comment (tomatosource) [reply main.go:33]: (wrap all errs, always lower case always verb first, never say error/failed/etc)

File: main.go [hunk 2]
```diff
@@ -6,60 +6,126 @@ import (
...
 	}
 	exPath := filepath.Dir(ex)
 	fmt.Println(exPath)
 
 	hostFlagPtr := flag.String("host", "localhost", "host to serve from, defaults to localhost")
 	portFlagPtr := flag.Int("port", 1234, "port to run the gopher server on, defaults to 1234")
-	serveDirFlagPtr := flag.String("directory", exPath+"/sample", "path of directory to serve, defaults to sample folder in CWD")
+	serveDirFlagPtr := flag.String(
```
Comment (tomatosource) [inline main.go:40]: forgot, but yeah i'd add a todo to use https://github.com/spf13/viper or something similar mainly to just explore 3rd party lib usage a bit :)

File: main.go [hunk 3]
```diff
@@ -6,60 +6,126 @@ import (
...
-		os.Exit(1)
+		return errors.Wrapf(err, "listening at %s", address)
 	}
-
 	defer l.Close()
 
+	// TODO
+	// setup sigterm/int/etc handler to stop accepting new conns and wait XXX time
+	// for active conns to close
+	// https://pkg.go.dev/os/signal#example_Notify
```
Comment (tomatosource) [inline main.go:63]: this is def worth playing with

File: main.go [hunk 4]
```diff
@@ -6,60 +6,126 @@ import (
...
+	scanner.Scan()
+	input := scanner.Bytes() // get bytes out after scanning
+
+	if len(input) == 0 {
+		entries, err := fs.ReadDir(os.DirFS(serveDir), ".")
+
+		if err != nil {
+			log.Println(err)
+			// TODO maybe pass an errs chan to conn handler for server to do something with?
+			// or maybe just write something back to the client conn here
```
Comment (tomatosource) [inline main.go:89]: not sure if it makes sense for the server to manage the conn errs, but either way shouldnt kill shit

File: main.go [hunk 5]
```diff
@@ -6,60 +6,126 @@ import (
...
 	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	ItemTypeTextFile  = 0
+	ItemTypeDirectory = 1
+	// TODO: add more
 )
```
Comment (tomatosource) [inline main.go:22]: just convention, consts at top of file, usually it goes imports, consts, vars, types, funcs

File: main.go [hunk 6]
```diff
@@ -117,55 +186,5 @@ func buildGopherStringForDirectoryEntries(directoryEntries []fs.DirEntry, select
 
 	gopherString += "." // append deliminating period
 
-	return gopherString
-}
-
-func handleConnection(c net.Conn, serveDir string, port int, host string) {
```
Comment (tomatosource) [inline main.go:123]: i regret reording funcs now :|

**Comment by tomatosource** [comment]
closing cos it fucks my queue

File: go.mod [hunk 7]
```diff
@@ -1,3 +1,5 @@
-module gorpeh
+module github.com/korziee/gorpeh
 
-go 1.19
+go 1.18
```
Comment (korziee) [inline go.mod:3]: why go 1.18 lol?

File: main.go [hunk 8]
```diff
@@ -6,60 +6,126 @@ import (
...
-	fmt.Println("serving dir: ", serveDir)
-	l, err := net.Listen("tcp", address)
+	log.Println("listening on: ", address)
+	log.Println("serving dir: ", serveDir)
 
+	l, err := net.Listen("tcp", address)
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		return errors.Wrapf(err, "listening at %s", address)
```
Comment (korziee) [inline main.go:56]: yeah this is much nicer, gives a lot more context on the error

File: main.go (hunk 4 above)
Comment (korziee) [reply main.go:89]: yeah good call, there is an error item type in gopher, I'll add that in one day

File: main.go (hunk 1 above)
Comment (korziee) [reply main.go:33]: I like this

**Pull request closed without merging**